compiler/damlc/daml-package-config/src/DA/Daml/Package/Config.hs

```diff
@@ -202,11 +253,16 @@ fullParseMultiPackageConfig :: ProjectPath -> IO MultiPackageConfigFields
 fullParseMultiPackageConfig startPath = do
   mpcs <- exploreAndFlatten startPath $ \projectPath -> do
     multiPackage <- readMultiPackageConfig projectPath
-    multiPackageConfigI <- either throwIO pure (parseMultiPackageConfig multiPackage)
+    multiPackageConfigI <- either throwIO pure (parseMultiPackageConfig multiPackage $ unwrapProjectPath projectPath)
     canonMultiPackageConfigI <- canonicalizeMultiPackageConfigIntermediate projectPath multiPackageConfigI
     pure (ProjectPath <$> mpiOtherConfigFiles canonMultiPackageConfigI, mpiConfigFields canonMultiPackageConfigI)
 
-  pure $ MultiPackageConfigFields $ nubOrd $ concatMap mpPackagePaths mpcs
+  pure MultiPackageConfigFields
+    { mpPackagePaths = nubOrd $ concatMap mpPackagePaths mpcs
+    , mpCompositeDars = mpCompositeDars $ head mpcs
+    , mpTransitiveCompositeDarNames = (\mpc -> (mpPath mpc, cdName <$> mpCompositeDars mpc)) <$> tail mpcs
+    , mpPath = mpPath $ head mpcs
```

Comment:
I'm a bit confused by `head` and `tail` here, is the idea that the composite dars are only defined at the multi package root?

Reply:
Yes exactly! We had some discussion about shadowing composite dars, and decided the cleanest thing we can do is only allow top level composite dars to be built, but still warn about shadows.